### PR TITLE
Add docs, release, and contributing to README

### PR DESCRIPTION
**Pull request by rkuykendall**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +33,14 @@
 #### yarn
 `yarn add --dev @mighty-justice/fields-ant`
 
-### Usage
+### Releasing new version
+
+- Release: `npm run deploy`
+
+This will run checks for `test`, `lint`, `build`, bump the version number,
+publish to npm and publish the docs to Github Pages.
```
Comment (alexandermeyer) [inline README.md]: Since Mighty endorses the Oxford comma, this should have a comma

File: docs/Components.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Components
+
+# Card
```
Comment (alexandermeyer) [inline docs/Components.md]: What is the ordering for this components file? Some decisions like `EditableArrayCard` coming before `ArrayCard` are confusing

File: docs/Interfaces.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Interfaces
+
+## FieldConfig
+
+This defines the core common interface for mighty fields, the IFieldConfig.
```
Comment (alexandermeyer) [inline docs/Interfaces.md]: Capital `M` in `Mighty`?

File: docs/Interfaces.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+This excludes all fields used only by a single type ( see below )
+It also will almost never be written-out by a user, since it's
```
Comment (alexandermeyer) [inline docs/Interfaces.md]: I believe `written-out` is actually `written out`

File: docs/Components.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
+# Components
```
Comment (alexandermeyer) [inline docs/Components.md]: These examples have the emphasis as `Components`. Shouldn't they have `##`?

File: README.md [hunk 6]
```diff
@@ -26,7 +33,14 @@
 #### yarn
 `yarn add --dev @mighty-justice/fields-ant`
 
-### Usage
+### Releasing new version
```
Comment (alexandermeyer) [inline README.md]: `a new version` or `new versions`

File: docs/Components.md (hunk 2 above)
Comment (rkuykendall) [reply docs/Components.md]: Yeah the ideal was that it was simple=>complex, more used => less used. Fixed that dependency ordering problem.

**Pull request merged**